src/js/popup.js

```diff
@@ -205,54 +249,63 @@ function send_error(message) {
 }
 
 /**
-* activate PB for site event handler
-*/
-function active_site() {
+ * activate PB for site event handler
+ */
+function activateOnSite() {
   $("#activate_site_btn").toggle();
   $("#deactivate_site_btn").toggle();
   $("#blockedResourcesContainer").show();
-  getTab(function(tab) {
-    badger.enablePrivacyBadgerForOrigin(backgroundPage.extractHostFromURL(tab.url));
-    badger.refreshIconAndContextMenu(tab);
-    reloadTab(tab.id);
+
+  chrome.runtime.sendMessage({
+    type: "activateOnSite",
+    tabHost: POPUP_DATA.tabHost,
+    tabId: POPUP_DATA.tabId,
+    tabUrl: POPUP_DATA.tabUrl
+  }, () => {
+    chrome.tabs.reload(POPUP_DATA.tabId);
     window.close();
   });
 }
 
 /**
-* de-activate PB for site event handler
-*/
-function deactive_site() {
+ * de-activate PB for site event handler
+ */
+function deactivateOnSite() {
   $("#activate_site_btn").toggle();
   $("#deactivate_site_btn").toggle();
   $("#blockedResourcesContainer").hide();
-  getTab(function(tab) {
-    badger.disablePrivacyBadgerForOrigin(backgroundPage.extractHostFromURL(tab.url));
-    badger.refreshIconAndContextMenu(tab);
-    reloadTab(tab.id);
+
+  chrome.runtime.sendMessage({
+    type: "deactivateOnSite",
+    tabHost: POPUP_DATA.tabHost,
+    tabId: POPUP_DATA.tabId,
+    tabUrl: POPUP_DATA.tabUrl
+  }, () => {
+    chrome.tabs.reload(POPUP_DATA.tabId);
     window.close();
   });
 }
 
 /**
-* Handler to undo user selection for a tracker
-*
-* @param e The object the event triggered on
-* @returns {boolean} false
-*/
+ * Handler to undo user selection for a tracker
+ *
+ * @param {Event} e The object the event triggered on
+ */
 function revertDomainControl(e) {
-  var tabId = parseInt($('#associatedTab').attr('data-tab-id'), 10);
   var $elm = $(e.target).parent();
   var origin = $elm.data('origin');
-  badger.storage.revertUserAction(origin);
-  var defaultAction = badger.storage.getBestAction(origin);
-  var selectorId = "#"+ defaultAction +"-" + origin.replace(/\./g,'-');
-  var selector = $(selectorId);
-  selector.click();
-  $elm.removeClass('userset');
-  reloadTab(tabId);
-  window.close();
-  return false;
+  chrome.runtime.sendMessage({
+    type: "revertDomainControl",
+    origin: origin
+  }, (response) => {
+    var defaultAction = response.action;
+    var selectorId = "#"+ defaultAction +"-" + origin.replace(/\./g,'-');
+    var selector = $(selectorId);
+    selector.click();
```

Comment:
the "revertDomainControl" message shouldn't need any information in the response, but the popup does need a way to know that the message has been processed so that it can trigger the page reload and close the popup.